src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -284,7 +288,7 @@ internal void SnapshotDeltaWithMetricPointReclaim()
             }
         }
 
-        for (int i = startIndexForReclaimableMetricPoints; i < this.CardinalityLimit; i++)
+        for (int i = startIndexForReclaimableMetricPoints; i < this.NumberOfReservedMetricPoints; i++)
```

Comment:
Now that we always reserve two MetricPoints, the above code should be update to always initialize `startIndexForReclaimableMetricPoints` as `2` instead of doing that conditionally on whether emit overflow is true.